cli/flox/src/commands/list.rs

```diff
@@ -105,8 +107,41 @@ impl List {
     /// print the manifest contents
     fn print_config(flox: &Flox, env: &mut ConcreteEnvironment, lockfile: &Lockfile) -> Result<()> {
         let is_composed = lockfile.compose.is_some();
+
+        // A visitor that converts inline tables to proper tables
+        // Nested tables are rendered as `dotted` tables.
+        // The default behavior when instantiating with `Visitor::new_for_document`,
+        // is to render toplevel tables as non-dotted, sections.
+        struct Visitor {
+            dotted: bool,
+        }
+        impl Visitor {
+            fn new_for_document() -> Self {
+                Visitor { dotted: false }
+            }
+        }
+        impl VisitMut for Visitor {
+            fn visit_table_like_kv_mut(&mut self, _key: KeyMut<'_>, node: &mut Item) {
+                if let toml_edit::Item::Value(Value::InlineTable(inline_table)) = node {
+                    let mut table = std::mem::take(inline_table).into_table();
+                    table.set_implicit(true);
+                    table.set_dotted(self.dotted);
+                    toml_edit::visit_mut::visit_table_mut(
+                        &mut Visitor { dotted: true },
+                        &mut table,
+                    );
+                    *node = toml_edit::Item::Table(table);
+                }
+            }
+        }
```

Comment:
blocking: needs at least some test coverage, although maybe what we have in list.bats is sufficient. Will be more clear from the diff once you fix the tests

Reply:
https://github.com/flox/flox/pull/2887